gef.py

```diff
@@ -2593,6 +2593,79 @@ def inner_f(*args, **kwargs):
     return wrapper
 
 
+def FakeExit(*args, **kwargs):
+    raise RuntimeWarning
+
+sys.exit = FakeExit
+
+def parse_arguments(required_arguments, optional_arguments):
+    """Argument parsing decorator."""
+
+    def int_wrapper(x): return int(x, 0)
+
+    def decorator(f):
+        def wrapper(*args, **kwargs):
+            parser = argparse.ArgumentParser(prog=args[0]._cmdline_, add_help=True)
+            for argname in required_arguments:
+                argvalue = required_arguments[argname]
+                argtype = type(argvalue)
+                if argtype == int:
+                    argtype = int_wrapper
+
+                if argname.startswith("-"):
+                    # optional args
+                    if argtype == bool:
+                        parser.add_argument(argname, action="store_true" if argvalue else "store_false")
+                    else:
+                        parser.add_argument(argname, type=argtype, required=True, default=argvalue)
+                else:
+                    if argtype in (list, tuple):
+                        nargs = '*'
+                        argtype = type(argvalue[0])
+                    else:
+                        nargs = '?'
+                    # positional args
+                    parser.add_argument(argname, type=argtype, default=argvalue, nargs=nargs)
+
+            for argname in optional_arguments:
+                if not argname.startswith("-"):
+                    # refuse positional arguments
+                    continue
+                argvalue = optional_arguments[argname]
+                argtype = type(argvalue)
+                if argtype == int:
+                    argtype = int_wrapper
+                if argtype == bool:
+                    parser.add_argument(argname, action="store_true" if argvalue else "store_false")
+                else:
+                    parser.add_argument(argname, type=argtype, default=argvalue)
+
+            _, cmd_args = args[0], args[1:]
+            parsed_args = parser.parse_args(*cmd_args)
+            kwargs["arguments"] = parsed_args
+            return f(*args, **kwargs)
+        return wrapper
+    return decorator
+
+
+def copy_to_clipboard(data):
+    """Helper function to submit data to the clipboard"""
+    if sys.platform == "linux":
+        xclip = which("xclip")
+        prog = [xclip, "-selection", "clipboard", "-i"]
+    elif sys.platform == "darwin":
+        pbcopy = which("pbcopy")
+        prog = [pbcopy]
+    else:
+        raise NotImplementedError("Unsupported OS")
```

Comment:
Let's clarify: something like "Don't know how to copy to clipboard on this OS" or similar.

Reply:
Nope, the issue is clearly that the OS used is not supported (check the conditions of the `if`).

But if you want, I can wrap the whole tests `if sys.platform == "linux":...` with a `try: ... except FileNotFoundError: err('Don't know how to copy to clipboard on this OS"')`

Reply:
Yes, but the error doesn't show that it has to do with copy and paste. I am saying to make that clear for people who don't see stack traces. Since this can be called from other commands it can be confusing.

Reply:
Ah! Then suggestion: we implement an exception like `NotSupportedOsError` or something, and then in the message we can refer to the fact it's coming from copy/paste function?

Reply:
that can work, might be heavy. I would be happy even with `raise NotImplementedError("paste: Unsupported OS")`

This is probably fine as is, I can just foresee people seeing 'unsupported os' without context about copy and paste and filing bugs and that getting annoying ;)

Reply:
Done